packages/editor-ui/src/components/NodeIcon.vue

```diff
@@ -1,34 +1,38 @@
 <template>
-	<div class="node-icon-wrapper" :style="iconStyleData">
-		<div v-if="nodeIconData !== null" class="icon">
-			<img v-if="nodeIconData.type === 'file'" :src="nodeIconData.fileBuffer || nodeIconData.path" :style="imageStyleData" />
-			<font-awesome-icon v-else :icon="nodeIconData.icon || nodeIconData.path" :style="fontStyleData" />
-		</div>
-		<div v-else class="node-icon-placeholder">
-			{{nodeType !== null ? nodeType.displayName.charAt(0) : '?' }}
-		</div>
-	</div>
+	<n8n-node-icon
+		:type="nodeIconData.type"
+		:path="nodeIconData.path || nodeIconData.icon || nodeIconData.fileBuffer"
```

Comment:
is path best name for this prop? or maybe split into multiple props?

Reply:
It's not 😁 I wanted to keep it simple by having one prop for icon, so maybe `iconSource` is a better name for it. Having two (or three) props for this would make it more readable, but I feel that there are already too many props in the component...

Reply:
After giving this a thought, I decided to split it into two props:
- `iconPath`: Which will be used to map `path` and `fileBuffer` and will be used for file icons,
-  `iconName`: Used for icon `type`, maps icon `prop`